android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -65,6 +68,48 @@ public TiListView(ListViewProxy proxy)
 		this.recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
 		this.recyclerView.setFocusableInTouchMode(false);
 
+		// Add listener to fire scroll events.
+		this.recyclerView.addOnScrollListener(new RecyclerView.OnScrollListener()
+		{
+			@Override
+			public void onScrollStateChanged(@NonNull RecyclerView recyclerView, int newState)
+			{
+				super.onScrollStateChanged(recyclerView, newState);
+
+				if (isScrolling && newState == RecyclerView.SCROLL_STATE_IDLE) {
+					isScrolling = false;
+					proxy.fireSyncEvent(TiC.EVENT_SCROLLEND, generateScrollPayload());
+				}
+			}
+
+			@Override
+			public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy)
+			{
+				super.onScrolled(recyclerView, dx, dy);
+
```

Comment:
I noticed that `onScrolled()` gets called on its own when the window opens. So I'm seeing "scrollstart" and "scrolling" events with "direction" set to `undefined` every time I open a window hosting the `ListView`.

I looked it up and it turns out `RecyclerView.onScroleld()` will be called when add/removing rows that are visibly onscreen. Maybe we should check if arguments `dx` and `dy` are zero before proceeding like the below? I haven't tried this so I'm not sure if this is right way to avoid it.
```java
if ((dx == 0) && (dy == 0)) {
    return;
}
```